backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java

```diff
@@ -2,58 +2,208 @@
-			return null;
-		}
-
-		try {
-			var phoneUtil = PhoneNumberUtil.getInstance();
-			return phoneUtil.format(phoneUtil.parse(userSuppliedPhoneNumber, "US"),
-					PhoneNumberUtil.PhoneNumberFormat.NATIONAL);
-		} catch (NumberParseException parseException) {
-			throw new IllegalGraphqlArgumentException("[" + userSuppliedPhoneNumber + "] is not a valid phone number.");
-		}
-	}
+    private static DateTimeFormatter US_SLASHDATE_SHORT_FORMATTER = DateTimeFormatter.ofPattern("M/d/yyyy");
+    private static final int MAX_STRING_LENGTH = 500;
+
+    public static final LocalDate parseUserShortDate(String d) {
+        String date = parseString(d);
+        if (date == null) {
+            return null;
+        }
+        try {
+            return LocalDate.parse(date, US_SLASHDATE_SHORT_FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new IllegalGraphqlArgumentException("[" + d + "] is not a valid date.");
+        }
+    }
+
+    public static final LocalDate parseUserDate(String userSuppliedDateString) {
+        if (userSuppliedDateString == null) {
+            return null;
+        }
+        if (userSuppliedDateString.contains("/")) {
+            return LocalDate.parse(userSuppliedDateString, US_SLASHDATE_FORMATTER);
+        } else {
+            return LocalDate.parse(userSuppliedDateString); // ISO_LOCAL_DATE is the default
+        }
+    }
+
+    public static final Date parseUserDateTime(String userSuppliedIsoDateString) {
+        if (userSuppliedIsoDateString == null) {
+            return null;
+        }
+        Instant isoDateInstant;
+        try {
+              isoDateInstant = Instant.parse(userSuppliedIsoDateString);
+        }
+        catch (DateTimeParseException parseException) {
+            throw new IllegalGraphqlArgumentException("[" + userSuppliedIsoDateString + "] is not a valid date.");
+        }
+          return Date.from(isoDateInstant);
+    }
+
+    public static String parsePhoneNumber(String userSuppliedPhoneNumber) {
+        if (userSuppliedPhoneNumber == null) {
+            return null;
+        }
+
+        try {
+            var phoneUtil = PhoneNumberUtil.getInstance();
+            return phoneUtil.format(phoneUtil.parse(userSuppliedPhoneNumber, "US"),
+                    PhoneNumberUtil.PhoneNumberFormat.NATIONAL);
+        } catch (NumberParseException parseException) {
+            throw new IllegalGraphqlArgumentException("[" + userSuppliedPhoneNumber + "] is not a valid phone number.");
+        }
+    }
+
+    public static String parseString(String value) {
+        if (value == null || "".equals(value)) {
+            return null;
+        }
+        if (value.length() >= MAX_STRING_LENGTH) {
+            throw new IllegalGraphqlArgumentException(
+                "Value received exceeds field length limit of " + MAX_STRING_LENGTH + " characters"
+            );
+        }
+        return value.trim();
+    }
+
+    public static UUID parseUUID(String uuid) {
```

Comment:
`Optional<UUID>` is the "nicer" way to do this, but may not be appropriate—just making notes as I go, here.